core/test/lib/libledger-test/MemPreferencesBackend.cpp

```diff
@@ -9,13 +9,15 @@ namespace ledger {
 
             std::experimental::optional<std::vector<uint8_t>> MemPreferencesBackend::get(const std::vector<uint8_t> & key)
             {
-               if (_data.find(key) != _data.end())
+                std::unique_lock<std::mutex> lock(_mtx);
+                if (_data.find(key) != _data.end())
                     return _data.at(key);
                 return optional<std::vector<uint8_t>>();
             }
 
             bool MemPreferencesBackend::commit(const std::vector<api::PreferencesChange> & changes)
             {
+                std::unique_lock<std::mutex> lock(_mtx);

```

Comment:
I would suggest to move this `lock` into the loop because `changes` does not need to be locked.

Reply:
done